eng/common/scripts/Update-DocsMsMetadata.ps1

```diff
@@ -230,17 +230,39 @@ function UpdateDocsMsMetadataForPackage($packageInfoJsonLocation) {
   Set-Content -Path $readmeLocation -Value $outputReadmeContent
 }
 
-# For daily update and release, validate DocsMS publishing using the language-specific validation function
-if ($ValidateDocsMsPackagesFn -and (Test-Path "Function:$ValidateDocsMsPackagesFn")) {
-  Write-Host "Validating the packages..."
+$allSucceeded = $true
+foreach ($packageInfoLocation in $PackageInfoJsonLocations) {
 
-  $packageInfos = @($PackageInfoJsonLocations | ForEach-Object { GetPackageInfoJson $_ })
+  if ($ValidateDocsMsPackagesFn -and (Test-Path "Function:$ValidateDocsMsPackagesFn")) {
+    Write-Host "Validating the packages..."
 
-  &$ValidateDocsMsPackagesFn -PackageInfos $packageInfos -PackageSourceOverride $PackageSourceOverride -DocValidationImageId $DocValidationImageId -DocRepoLocation $DocRepoLocation
-}
+    $packageInfo =  GetPackageInfoJson $packageInfoLocation
+    # "Validate-${Language}-DocMsPackages"
+    $isValid = &$ValidateDocsMsPackagesFn `
+      -PackageInfos $packageInfo `
+      -PackageSourceOverride $PackageSourceOverride `
+      -DocValidationImageId $DocValidationImageId `
+      -DocRepoLocation $DocRepoLocation
+
+    if (!$isValid) {
+      Write-Host "Package validation failed for package: $packageInfoLocation"
+      $allSucceeded = $false
+
+      # Skip the later call to UpdateDocsMsMetadataForPackage because this 
+      # package has not passed validation
+      continue
+    }
+  }
 
-foreach ($packageInfoLocation in $PackageInfoJsonLocations) {
   Write-Host "Updating metadata for package: $packageInfoLocation"
   # Convert package metadata json file to metadata json property.
   UpdateDocsMsMetadataForPackage $packageInfoLocation
 }
+
+# Set a variable which will be used by the pipeline later to fail the build if
+# any packages failed validation
+if ($allSucceeded) {
```

Comment:
Why not just exit 1 or 0 here instead of passing this out?

Reply:
Two arguments for how it's writ: 

1. One package failing should not block other valid packages from daily docs previews, and failure would end the pipeline and stop the [changes from being pushed](https://github.com/Azure/azure-sdk-for-java/blob/main/eng/common/pipelines/templates/steps/update-docsms-metadata.yml#L109)
2. 1. Jobs like Java's nightly release also includes a process to export text files (this should be done differently in the future) https://github.com/Azure/azure-sdk-for-java/blob/main/eng/pipelines/templates/stages/archetype-java-release.yml#L476

Reply:
The parameter is allSucceeded so isn't it running all of the validation already? Or is this script called multiple times? If it is called multiple times then your exporting of this variable will only work for the last run and will overwrite any previous failures.

Reply:
`Update-DocsMsMetadata.ps1` is called [once](https://github.com/Azure/azure-sdk-for-java/blob/main/eng/common/pipelines/templates/steps/update-docsms-metadata.yml#L94) with an array of the package JSON locations it's validating. That script validates those packages in a loop and then can return this aggregate result.

Reply:
Once it returns that result why wouldn't this script fail with that aggregate results? Are there other steps after the running of this script that you still need to run? This seems like a round-about way of accomplishing that goal.

Reply:
`docsms-ensure-validation.yml` Is added after the `git-push-changes.yml` to enable us to push changes for packages that passed validation but still fail the pipeline after those packages have been pushed.

Reply:
OK so this is all about having a continueOnError setting while still ensuring some actions are taken. I guess we can live with this but it definitely seems like a roundabout way.